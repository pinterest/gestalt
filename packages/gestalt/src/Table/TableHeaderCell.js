// @flow strict
import * as React from 'react';
import styles from './Table.css';

type Props = {|
  children: React.Node,
  colSpan?: number,
  rowSpan?: number,
  scope?: 'col' | 'colgroup' | 'row' | 'rowgroup',
|};

export default function TableHeaderCell(props: Props) {
  const { children, colSpan, scope, rowSpan } = props;

  return (
    <th
      className={styles.th}
      scope={scope || 'col'}
      colSpan={colSpan}
      rowSpan={rowSpan}
    >
      {children}
    </th>
  );
}
