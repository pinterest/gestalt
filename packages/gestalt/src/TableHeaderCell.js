// @flow strict
import React, { type Node } from 'react';
import styles from './Table.css';

type Props = {|
  children: Node,
  colSpan?: number,
  rowSpan?: number,
  scope?: 'col' | 'colgroup' | 'row' | 'rowgroup',
|};

export default function TableHeaderCell(props: Props): Node {
  const { children, colSpan, scope, rowSpan } = props;

  return (
    <th className={styles.th} scope={scope || 'col'} colSpan={colSpan} rowSpan={rowSpan}>
      {children}
    </th>
  );
}
