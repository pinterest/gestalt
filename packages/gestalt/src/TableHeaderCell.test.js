// @flow strict
import renderer from 'react-test-renderer';
import TableHeaderCell from './TableHeaderCell';

test('renders correctly', () => {
  const tree = renderer.create(<TableHeaderCell>column name</TableHeaderCell>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with props', () => {
  const tree = renderer
    .create(
      <TableHeaderCell colSpan={3} rowSpan={2} scope="row">
        row name
      </TableHeaderCell>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
