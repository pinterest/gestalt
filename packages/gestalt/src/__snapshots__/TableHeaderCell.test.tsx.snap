// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<th
  className="th"
  scope="col"
  style={
    Object {
      "left": undefined,
      "right": undefined,
    }
  }
>
  column name
</th>
`;

exports[`renders correctly with props 1`] = `
<th
  className="th"
  colSpan={3}
  rowSpan={2}
  scope="row"
  style={
    Object {
      "left": undefined,
      "right": undefined,
    }
  }
>
  row name
</th>
`;
